common/operations.go

```diff
@@ -7,52 +7,78 @@ import (
 
 const (
 
-	// Operation Commands
-	BookInfoCommand  = "bookinfo"
-	HTTPBinCommand   = "httpbin"
-	ImageHubCommand  = "imagehub"
-	EmojiVotoCommand = "emojivoto"
+	// Common Operations
+	BookInfoOperation  = "bookinfo"
+	HTTPBinOperation   = "httpbin"
+	ImageHubOperation  = "imagehub"
+	EmojiVotoOperation = "emojivoto"
 
-	// Validate Operation Commands
-	ValidateSmiConformance = "validate_smi_conformance_test"
+	// Validate Operations
+	SmiConformanceOperation = "smi_conformance"
 
-	// Custom Operation Commands
-	CustomOpCommand = "custom"
+	// Custom Operation
+	CustomOperation = "custom"
+
+	// Additional Properties
+	ServiceName = "service_name"
 )
 
 var (
 	Operations = adapter.Operations{
-		BookInfoCommand: &adapter.Operation{
+		BookInfoOperation: &adapter.Operation{
 			Type:        int32(meshes.OpCategory_SAMPLE_APPLICATION),
 			Description: "Istio Book Info Application",
 			Versions:    adapter.NoneVersion,
-			Template:    "templates/bookinfo.yaml",
+			Templates: []adapter.Template{
+				"https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/platform/kube/bookinfo.yaml",
```

Comment:
Well referencing directly Is not possible but loop would help, yes. I didn't want to make the object more complex so went for an array, we can switch to struct if u think its a string requirement.

Reply:
Hmm not quite sure we're referring to the same thing (a disadvantage of asynchronous written communication), and not quite sure about the reference to loop here.

The version I referred to in this comment was referring to the version of Istio (and others) instead of master, e.g. https://raw.githubusercontent.com/istio/istio/1.6.1/samples/bookinfo/platform/kube/bookinfo.yaml` 

The comment about the link between the array of versions and array of templates is of course related. I agree that we should not make things (too) complex unless there is a need, and not anticipate future refinements. In this case however I think it would help to understand what version is used for, and if I understand correctly its implementation in the API is just around the corner.. :smile: 

So I think it would be the right thing to do.

Something to the effect of (naming just a suggestion to illustrate):

```
type Template string

type OperationVersion struct {
	Templates [] Template
}

type Versions map[string]OperationVersion

versions := Versions{
	"1.6.1": OperationVersion{
		Templates: []Template{"https://raw.githubusercontent.com/istio/istio/1.6.1/samples/bookinfo/platform/kube/bookinfo.yaml"},
	},
	"default": OperationVersion{
		Templates: []Template{"https://raw.githubusercontent.com/istio/istio/1.6.13/samples/bookinfo/platform/kube/bookinfo.yaml"},
	},
	"latest": OperationVersion{
		Templates: []Template{"https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/platform/kube/bookinfo.yaml"},
	},
}

```